code_folder/A1_csv_to_json.py

```diff
@@ -0,0 +1,343 @@
+    then converted to a dictionary; the name of the csv file is used as the
+    main key of the dictionary. Exceptions are made for the files
+    ["economic_data", "project", "project_data", "simulation_settings"], here
+    no main key is added. Another exception is made for the file
+    "energyStorage". When this file is processed, the according "storage_"
+    files (names of the "storage_"-columns in "energyStorage" are called and
+    added to the energyStorage Dictionary.
+
+
+    :param input_directory: str
+        path of the directory where the input csv files can be found
+    :param filename: str
+        name of the inputfile that is transformed into a json, without
+        extension
+    :param parameters: list
+        List of parameters names that are required
+    :return: dict
+        the converted dictionary
+    """
+
+    logging.debug(
+        "Loading input data from csv: %s", filename
+    )
+    csv_default_directory=os.path.join(input_directory, "default_csv/")
+    df = pd.read_csv(
+        os.path.join(input_directory, "csv/", "%s.csv" % filename),
+        sep=",",
+        header=0,
+        index_col=0,
+    )
+
+    # check parameters
+    extra = list(set(parameters) ^ set(df.index))
+    if len(extra) > 0:
+        for i in extra:
+            if i in parameters:
+                logging.error(
+                    "In the file %s.csv" % filename
+                    + " the parameter "
+                    + str(i)
+                    + " is missing. "
+                    "check %s", csv_default_directory + "for correct "
+                    "parameter names."
+                )
+            else:
+                logging.error(
+                    "In the file %s.csv" % filename
+                    + " the parameter "
+                    + str(i)
+                    + " is not recognized. \n"
+                    "check %s", csv_default_directory + "for correct "
+                    "parameter names."
+                )
+
+    # convert csv to json
+    single_dict2 = {}
+    single_dict = {}
+    for column in df:
+        if column != "unit":
+            column_dict = {}
+            for i, row in df.iterrows():
+                if row["unit"] == "str":
+                    column_dict.update({i: row[column]})
+                else:
+                    column_dict.update({i: {"value": row[column], "unit": row["unit"]}})
+            single_dict.update({column: column_dict})
+            # add exception for energyStorage
+            if filename == "energyStorage":
+                storage_dict = add_storage(column, input_directory)
+                single_dict[column].update(storage_dict)
+    # add exception for single dicts
+    if filename in ["economic_data", "project", "project_data", "simulation_settings"]:
+        return single_dict
+    elif "storage_" in filename:
+        return single_dict
+    else:
+        single_dict2.update({filename: single_dict})
```

Comment:
To print asset name list per csv:
```suggestion
        single_dict2.update({filename: single_dict})
        logging.info('From file %s following assets are added to the energy system: %s', filename, asset_name_string[:-2])
```

Reply:
done